spec/lib/chess/core/pieces/base_piece_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'spec_helper'
+
+RSpec.describe Chess::Core::Pieces::BasePiece do
+  subject(:piece) do
+    described_class.create_and_occupy(%i[R N B Q K P], [Chess::WHITE_TEAM, Chess::BLACK_TEAM].sample, cell)
+  end
+
+  let(:cells) do
+    cells = []
+
+    8.times do |row_index|
+      cells << []
+      8.times do |column_index|
+        name = "#{%w[a b c d e f g h][column_index]}#{row_index + 1}"
+        cells[row_index] << Chess::Core::Cell.new(name, Chess::WHITE_TEAM)
+      end
+    end
+
+    cells
+  end
+  let(:cell) do
+    cells[(Chess::MIN_INDEX..Chess::MAX_INDEX).to_a.sample][(Chess::MIN_INDEX..Chess::MAX_INDEX).to_a.sample]
```

Comment:
```suggestion
    cells.sample.sample
```

Reply:
I like a sample, we don't need to be deterministic here. That's the case for specific pieces